server/models.py

```diff
@@ -306,6 +309,8 @@ def can(cls, obj, user, action):
         is_staff = user.is_enrolled(obj.course.id, STAFF_ROLES)
         if action == "view":
             return is_staff or obj.visible
+        if action == "publish":
+            return is_staff and obj.visible
```

Comment:
Does it make sense to allow staff to be able to pre publish scores (even if it's marked as not visible to students)?